HTML table listing known issues.

jovirkku opened the pull request:
JB#57937

jovirkku commented:
New approach to PR #82

vigejolla reviewed (approved):
Too bad the github preview completely chokes on the html. But it is what it is and the end result seems to work, so approving.

sledges reviewed (changes requested):
To make GitHub preview work, all empty lines should be removed.

\<head> and <style> tags are not picked up by MarkDown (also I believe they shouldn't be within the \<table> scope), and #myTable isn't referenced in the table itself.

Once preview in GitHub is working, the next thing is to check how it's rendered in final GitHub pages. Then we can clearer see how to proceed with style and div customisations.

sledges commented:
This is the preview of the table - https://sledges.github.io/Releases/Known_Issues/ (instructions [here](https://github.com/sailfishos/docs.sailfishos.org/blob/master/README.md#preview-via-github-pages)), and after manually removing \<style>. \<head>. and \<div> the empty space above table header disappears, and table's formatting stays the same.

jovirkku commented:
The intended structure is:  Releases > Known_Issues, which holds the table of issues.
What makes the Known_Issues sub dir appear? It is not there at least yet.

vigejolla commented:
> The intended structure is: Releases > Known_Issues, which holds the table of issues. What makes the Known_Issues sub dir appear? It is not there at least yet.

I was going to mention that, but then I thought it was intended so I didn't. You need to remove the nav_exclude line from Known_Issues/README.md, and instead have "nav_order: 100" there. And Releases/README.md probably needs to have "has_children: true" but I'm not sure about that.

sledges commented:
> The intended structure is: Releases > Known_Issues, which holds the table of issues. What makes the Known_Issues sub dir appear? It is not there at least yet.

If you want to see the page and its sub-pages on the left navigation menu, you need to place them under one of the 4 main pillars of documentation (Develop, Reference, Services, Tools).

This is unfortunately how Just-the-Docs was designed (1 root document), and then I've extended it to 4 (https://github.com/sailfishos/docs.sailfishos.org/commit/0e8cad78925b99cdf5953070515e20f27bdd04d1)

There is possibility to add more such "pillars" like Releases etc, but then they will also appear in the top header, unless more code is written to explicitly hide them.

sledges commented:
> This is the preview of the table - https://sledges.github.io/Releases/Known_Issues/

To fit the table nicer, removing all mentions of "Vanha Rauma" from the 3rd column should widen the 2nd column automatically.

sledges commented on Releases/Known_Issues/README.md:
```diff
@@ -0,0 +1,245 @@
+          <td>Issues with mobile data persist on some SIM cards. Turn the Flight mode on and off to reset the network setup. Reverting the device to Android and re-installing Sailfish X has often helped. See our <a href="https://jolla.zendesk.com/hc/en-us/articles/115004283713">support article</a>.</td>
+          <td></td>
+          <td>Not planned</td>
+        </tr>        
```
nitpick: extra trailing spaces (also until the end of the file after other \</tr> tags)

sledges commented on Releases/Known_Issues/README.md:
```diff
@@ -0,0 +1,245 @@
+          <td>Issues with mobile data persist on some SIM cards. Turn the Flight mode on and off to reset the network setup. Reverting the device to Android and re-installing Sailfish X has often helped. See our <a href="https://jolla.zendesk.com/hc/en-us/articles/115004283713">support article</a>.</td>
+          <td></td>
+          <td>Not planned</td>
+        </tr>        
```
nitpick: extra trailing spaces (also until the end of the file after other </tr> tags)

```suggestion
        </tr>
```

jovirkku replied:
I do not see any harm in the preview...

sledges replied:
It's about longevity and ease of editing the source code:) https://softwareengineering.stackexchange.com/a/121560

jovirkku replied:
In the 3rd commit, I removed the trailing spaces.

jovirkku commented:
In my 2nd commit, I cleaned up the file in the way Simonas suggested. I also changed the nav tags according to Ville's proposal (but I did not touch the README at /Releases).

The new file can be viewed at https://github.com/jovirkku/docs.sailfishos.org/tree/a0562c2e17540b131b1ee62aa71c769233b2e277/Releases/Known_Issues

I would like to keep the release nicknames ("Vanha Rauma" and the like) as many users remember them better than the numbers. The numeric ID and the nickname are now folded to 2-3 rows, keeping the 3rd column reasonably narrow.

sledges commented on Releases/Known_Issues/README.md:
```diff
@@ -0,0 +1,245 @@
+        <tr>
+          <td>Xperia X, Jolla Tablet</td>
+          <td>Manual Android app installation from the terminal or file managers is currently not working on the Android 4.4 version of Android App Support on SailfishOS 4.4. To solve this, disable sandboxing for this handler by editing the file <code>/usr/share/applications/apkd-mime-handler.desktop</code> and adding the following to the end of the file:
+            <pre><code>
+[X-Sailjail]
+Disabled
+            </code></pre>
+          This doesn’t affect installing Android apps from the Jolla store, nor other Android stores.
+          </td>
```
\<pre> causes extra 2 newlines to appear surrounding the codeblock, because of how it's written. Can either write `<pre><code>[X-Sailjail]...` but that kills the readability, or just use MarkDown code-block like throughout the rest of documentation, also benefiting from syntax highlighter.
```suggestion

```ini
[X-Sailjail]
Disabled
\```
This doesn’t affect installing Android apps from the Jolla store, nor other Android stores.
</td>
```

Note: there is explicit newline before `...ini`, and `This ...` and `</td>` must not be indented in such cases.

Please remove the back-slash before the last back-tick triplet, I cannot suggest this better through GitHub:)

jovirkku replied:
I tried this but it does not seem to work as the preview has:

_adding the following to the end of the file: '''ini [X-Sailjail] Disabled \''' This doesn’t..._

sledges replied:
> Note: there is explicit newline before ...ini, and This ... and </td> must not be indented in such cases.

You must have overlooked that there needs to be an empty newline before the ```ini line.

jovirkku replied:
No. I have an empty line (e.g. #37) just above the line with three backticks (```) and "ini".  I have also consulted this page but in vain: https://www.markdownguide.org/extended-syntax/#fenced-code-blocks

sledges replied:
Sorry it worked for me on GitHub preview, but was mis-rendered on GitHub Pages. I've cleaned up my suggestion in the [next change request](https://github.com/sailfishos/docs.sailfishos.org/pull/112#pullrequestreview-1111926142).

Backslash needs to be removed always, GitHub shot themselves in the foot by not having a back-tick escape character for the "Suggested changes" feature:)

Also please don't forget to address the 2nd codeblock `/usr/bin/killall minimediaservice` below.

sledges commented on Releases/Known_Issues/README.md:
```diff
@@ -0,0 +1,245 @@
+        <tr>
+          <td>Jolla C</td>
+          <td>If there are issues with the camera of Jolla C, do the following:
+          <pre><code>
```
same as above

sledges commented on Releases/Known_Issues/README.md:
```diff
@@ -0,0 +1,245 @@
+          <td>Xperia 10 III</td>
+          <td>FM radio is not supported</td>
+          <td>4.4.0.64<br />Vanha Rauma</td>
+          <td>---</td>
```
Is there a difference between `---` and `Not planned`?

sledges replied:
And what do the empty cells mean in the light of the above?

jovirkku replied:
I will remove the triple dashes. The cells of the 4th column will either be empty or contain "not planned".

jovirkku commented on Releases/Known_Issues/README.md:
```diff
@@ -0,0 +1,245 @@
+        </tr>
+        <tr>
+          <td>Xperia X, Jolla Tablet</td>
+          <td>Manual Android app installation from the terminal or file managers is currently not working on the Android 4.4 version of Android App Support on SailfishOS 4.4. To solve this, disable sandboxing for this handler by editing the file <code>/usr/share/applications/apkd-mime-handler.desktop</code> and adding the following to the end of the file:
```
I'll fix that typo.

jovirkku commented on Releases/Known_Issues/README.md:
```diff
@@ -0,0 +1,245 @@
+        <tr>
+          <td>Xperia X, Jolla Tablet</td>
+          <td>Manual Android app installation from the terminal or file managers is currently not working on the Android 4.4 version of Android App Support on SailfishOS 4.4. To solve this, disable sandboxing for this handler by editing the file <code>/usr/share/applications/apkd-mime-handler.desktop</code> and adding the following to the end of the file:
+            <pre><code>
+[X-Sailjail]
+Disabled
+            </code></pre>
```
Thanks. That div trick seems to work. I left the backslash out from line 42. Finally. Mixing up html and md does not seem to be straightforward. But now we know how it goes.

sledges replied:
Where could I see those notes?

sledges replied:
> No other indents required here, just manually removing the backslash (backslash is only needed here to correctly render the "Suggested change" in this pull request).

Note to self and anyone reading: a triple-backtick-fenced code block can still be shown correctly if enclosed within a quadruple-backtick fence (https://stackoverflow.com/a/42320033):
`````
````suggestion
A suggested fenced code block:
```
echo "Hello world"
```
````
`````
Thus no pesky backslashes needed to render this:)

jovirkku commented:
@sledges: Please review again.

sledges commented on Releases/Known_Issues/README.md:
````diff
@@ -0,0 +1,254 @@
+Disabled
+```
+</div>
+
+This doesn’t affect installing Android apps from the Jolla store, nor other Android stores.
+</td>
````
Newline here not needed (only confuses otherwise), also no need to change indentation for normal text and \</td>: 
```suggestion
          This doesn’t affect installing Android apps from the Jolla store, nor other Android stores.
          </td>
```

sledges commented on Releases/Known_Issues/README.md:
````diff
@@ -0,0 +1,254 @@
+/usr/bin/killall minimediaservice
+```
+</div>
+
+and then start the camera again.
````
Same as earlier:
```suggestion
          and then start the camera again.
```

sledges commented on Releases/Known_Issues/README.md:
````diff
@@ -0,0 +1,254 @@
+          <td>If there are issues with the camera of Jolla C, do the following:
+<div markdown="1">
+
+```ini
````
This snippet is not an ini, but a shell script. However we do not highlight those (reasons [here](https://github.com/sailfishos/docs.sailfishos.org/blob/master/README.md)), thus `nosh`:)
```suggestion
```nosh
```

jovirkku replied:
Huoh, I didn't propose "ini". It was you, right?

sledges replied:
Allow me to explain: there are two codeblocks in the table. The first one is `ini`, the 2nd one is a shell command, thus `nosh` for its syntax highlighter.
My comment here is for the 2nd codeblock, please see the addressed line number/context.

sledges commented:
It will be time to squash all commits into one. Due to generic practice of forming commit message (should be short, and action-like e.g. "Add ..."), and portability, I suggest to rewrite the main commit message as:

```
Add HTML table which lists known issues

New approach to the work in https://github.com/sailfishos/docs.sailfishos.org/pull/82.

JB#57937
```

jovirkku commented:
I believe I have squashed all of my 9 commits into one now and used @sledge's proposal as the commit message for them.

sledges commented on Releases/Known_Issues/README.md:
```diff
@@ -0,0 +1,252 @@
+---
+title: Known issues in OS releases
+permalink: Releases/Known_Issues/
+parent: Known_Issues
+grand_parent: Releases
```
There is no grandparent for this page.
```suggestion
parent: Releases
```

sledges commented on Releases/Known_Issues/README.md:
````diff
@@ -0,0 +1,252 @@
+          <td>Manual Android app installation from the terminal or file managers is currently not working on the Android 4.4 version of Android App Support on Sailfish OS 4.4. To solve this, disable sandboxing for this handler by editing the file <code>/usr/share/applications/apkd-mime-handler.desktop</code> and adding the following to the end of the file:
+<div markdown="1">
+
+```nosh
````
This snippet is from an `ini` type of file. The 2nd code snippet at the bottom of the table is `nosh`. I hope this clears things up:)
```suggestion
```ini
```

sledges reviewed (approved):
Fantastic! Thank you for your first big contrib and patience.
We've now set the clean precedent for those who dare to deep dive into mixing Markdown and HTML:) and also for swifter editing of this very table in the future.

Please squash and I'll merge this.
